### PR TITLE
Upgrade to current hapi.js standards

### PR DESCRIPTION
**Pull request by sericaia**
Solves https://github.com/hapijs/bossy/issues/46

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 
 node_js:
-  - 0.10
-  - 0.12
-  - iojs
+  - 4.0
+  - 4
+  - 5
```
Comment (hueniverse) [inline .travis.yml]: Just `4` and `latest`

File: lib/index.js [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
+    const keys = {};
     definition = Joi.validate(definition, Schemas.definition).value;
     options = options || {};
 
-    var names = Object.keys(definition);
-    for (var i = 0, il = names.length; i < il; ++i) {
-        var name = names[i];
-        var def = Hoek.clone(definition[name]);
+    const names = Object.keys(definition);
+    for (let i = 0, il = names.length; i < il; ++i) {
```
Comment (hueniverse) [inline lib/index.js]: we no longer use the `il` convention.

File: lib/index.js [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
+        const name = names[i];
+        const def = Hoek.clone(definition[name]);
         def.name = name;
         keys[name] = def;
         if (def.alias) {
-            for (var a = 0, al = def.alias.length; a < al; ++a) {
+            for (let a = 0, al = def.alias.length; a < al; ++a) {
```
Comment (hueniverse) [inline lib/index.js]: If you update to the latest lab and code, this will error.

**Comment by sericaia** [comment]
@hueniverse is it fine now?

**Pull request merged**